larsim/EventGenerator/SingleGen_module.cc

```diff
@@ -447,11 +447,20 @@ namespace evgen {
     , fHistFileName(config().HistogramFile())
     , fPHist(config().PHist())
     , fThetaXzYzHist(config().ThetaXzYzHist())
-    , fEngine(createEngine(0))
+    , fEngine(art::ServiceHandle<rndm::NuRandomService>()->registerAndSeedEngine(createEngine(0),
+                                                                                 "HepJamesRandom",
+                                                                                 "Seed"))
```

Comment:
Why not to use [`registerAndSeedEngine(engine_t& engine, std::string type, SeedAtom const&`](https://code-doc.larsoft.org/docs/latest/html/classrndm_1_1NuRandomService.html#a70a3880022efe4b02b356b0dd84ce0d5):
```suggestion
    , fEngine(art::ServiceHandle<rndm::NuRandomService>()->registerAndSeedEngine(createEngine(0),
                                                                                 "HepJamesRandom",
                                                                                 config().Seed))
```
or [the variant with `std::optional` seed](https://code-doc.larsoft.org/docs/latest/html/classrndm_1_1NuRandomService.html#aed73c4bd19a744678a37579e5b433b65):
```suggestion
    , fEngine(art::ServiceHandle<rndm::NuRandomService>()->registerAndSeedEngine(createEngine(0),
                                                                                 "HepJamesRandom",
                                                                                 "",
                                                                                 config().Seed()))
```
In this way the initialisation of the seed should happen at once.